flight/Libraries/math/misc_math.h

```diff
@@ -100,6 +102,312 @@ static inline int16_t sin_approx(int32_t x)
 	return x * ( (3<<s_qP) - (x*x>>qR) ) >> qS;
 }
 
+/* Multiplies out = a * b
+ *
+ * Matrices are stored in row order, that is a[i*cols + j]
+ *
+ * output is arows by bcols
+ * a is arows by acolsbrows
+ * b is acolsbrows by bcols
+ */
+static inline void matrix_mul(const float *a, const float *b,
+		float *out, int arows, int acolsbrows, int bcols)
+{
+	const float * restrict apos = a;
+	float * restrict opos = out;
+
+	for (int ar = 0; ar < arows; ar++) {
+		for (int bc = 0 ; bc < bcols; bc++) {
+			float sum = 0;
+
+			const float * restrict bpos = b + bc;
+
+			int acbr;
+
+			for (acbr = 0; acbr < (acolsbrows & (~3)); acbr += 4) {
```

Comment:
Yah, I'll explain the unrolling in a code comment.

This is really nice-- unexpectedly so, even-- when the size is unknown, it unrolls it pretty much as written.  When the size is known and it's built with -Os, it encourages the implementation to explicitly unroll to just a slightly higher size than it would otherwise (like 8-10).